### PR TITLE
Delete confirmations on user pref, not always

### PR DESCRIPTION
**Pull request by chasehult**

### CODE REVIEW

File: tsutils/user_interaction.py [hunk 1]
```diff
@@ -22,7 +23,8 @@ async def send_repeated_consecutive_messages(ctx, message: str) -> discord.Messa
 
 async def get_user_confirmation(ctx, text: str,
                                 yes_emoji: SendableEmoji = YES_EMOJI, no_emoji: SendableEmoji = NO_EMOJI,
-                                timeout: int = 10) -> Literal[True, False, None]:
+                                timeout: int = 10, force_delete: bool = None, show_feedback: bool = False) \
```
Comment (RheingoldRiver) [inline tsutils/user_interaction.py:26]: `force_delete: Optional[bool]` right?

Comment (chasehult) [reply tsutils/user_interaction.py:26]: No, it's closer to Literal[True, False, None], but the input should never be None.

File: tsutils/user_interaction.py [hunk 2]
```diff
@@ -40,11 +42,29 @@ def check(reaction, user):
     except asyncio.TimeoutError:
         ret = None
 
-    await msg.delete()
+    if force_delete is not False and (force_delete is True
```
Comment (RheingoldRiver) [inline tsutils/user_interaction.py:45]: this logic would be clearer with an intermediate variable do_deletion

```
do_deletion = force_delete
if force_delete is None:
    do_deletion = await get_user_preference(ctx.bot, ctx.author, 'delete_confirmation', True)
```

Comment (RheingoldRiver) [reply tsutils/user_interaction.py:45]: actually idk what the last variable is, if that's a fallback it should be False

File: tsutils/user_interaction.py (hunk 1 above)
Comment (chasehult) [reply tsutils/user_interaction.py:26]: yeah Optional isn't right here imo

File: tsutils/user_interaction.py (hunk 2 above)
Comment (RheingoldRiver) [reply tsutils/user_interaction.py:45]: actually idk what the last param is, if it's a fallback it should be False

File: tsutils/user_interaction.py (hunk 1 above)
Comment (RheingoldRiver) [reply tsutils/user_interaction.py:26]: doesnt Optional mean that providing the param is optional

Comment (RheingoldRiver) [reply tsutils/user_interaction.py:26]: yeah i asked max he says Optional is right

**Pull request merged**